### PR TITLE
[DECOUPLED-MODE] Update some unit tests for accommodating decoupled mode

### PR DESCRIPTION
**Pull request by gulsumgudukbay**
# Description
This PR updates newly added tests to make them accommodate decoupled mode. 

# Tests
Tested the UTs locally and they work on decoupled mode. The changes only involve decoupled mode, and does not disturb regular mode behavior.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: tests/utils/test_helpers.py [hunk 1]
```diff
@@ -76,9 +76,21 @@ def get_test_base_output_directory(cloud_path=None):
   return cloud_path or "gs://runner-maxtext-logs"
 
 
+def get_test_config_path_for(relative_path: str):
```
Comment (shralex) [inline tests/utils/test_helpers.py]: Can we merge this into get_test_config_path ? Here's what Gemini suggests: 

get_test_config_path(relative_path="base.yml"):
    Returns the absolute path for a test config.
    
    If relative_path is 'base.yml', it applies the decoupled-mode logic.

    if relative_path == "base.yml":
        base_cfg = "decoupled_base_test.yml" if is_decoupled() else "base.yml"
        return os.path.join(MAXTEXT_CONFIGS_DIR, base_cfg)
    return os.path.join(MAXTEXT_CONFIGS_DIR, relative_path)

File: tests/unit/attention_test.py [hunk 2]
```diff
@@ -1285,7 +1285,10 @@ def test_projection_initialization(self):
     mla_extra_args = {"ici_fsdp_parallelism": jax.device_count()} if is_decoupled() else {}
     mla_config_args.update(mla_extra_args)
     _, mla_layer = self.init_mla(mla_config_args, rope_type="default")
-    _, mla_layer = self.init_mla(self.config_arguments, rope_type="default")
+    # Also test the baseline init path; in decoupled mode inject mesh override.
+    mla_baseline_args = self.config_arguments.copy()
+    mla_baseline_args.update(mla_extra_args)
+    _, mla_layer = self.init_mla(mla_baseline_args, rope_type="default")
```
Comment (shralex) [inline tests/unit/attention_test.py]: is this now redundant with the mla_layer initialization above ? can we keep one of them ?

File: tests/unit/mhc_test.py [hunk 3]
```diff
@@ -92,8 +92,10 @@ class TestMHC(unittest.TestCase):
 
   def setUp(self):
     self.dim = 16
+    extra_args = {"ici_fsdp_parallelism": jax.device_count()} if is_decoupled() else {}
```
Comment (SurbhiJainUSC) [inline tests/unit/mhc_test.py]: This seems to be repeated in multiple test files. Can we add a helper method in `tests/utils/test_helpers.py` to remove this duplication?

**Comment by SurbhiJainUSC** [comment]
Can you resolve the conflicts?

**Comment by gulsumgudukbay** [comment]
> Can you resolve the conflicts?

I rebased and resolved the conflicts

File: tests/integration/train_tests.py [hunk 4]
```diff
@@ -18,11 +18,17 @@
 import pytest
 import jax
 
+from jax._src import test_util as jtu
```
Comment (shralex) [inline tests/integration/train_tests.py]: is jax._src the best source for this ? is there a public API ?

File: tests/utils/test_helpers.py [hunk 5]
```diff
@@ -24,15 +24,50 @@
 from maxtext.utils.globals import MAXTEXT_CONFIGS_DIR
 
 
-def get_test_config_path():
-  """Return absolute path to the chosen test config file.
+def get_test_config_path(relative_path: str = "base.yml"):
```
Comment (shralex) [inline tests/utils/test_helpers.py:27]: This automatically overrides base.yml with decoupled_base_test.yml when is_decoupled() is true, lets make sure this behavior is well documented in base.yml and docs.

File: tests/utils/test_helpers.py [hunk 6]
```diff
@@ -24,15 +24,50 @@
...
+
+    overrides = {"ici_fsdp_parallelism": jax.device_count() if fsdp_parallelism is None else int(fsdp_parallelism)}
+    if include_mesh_defaults:
+      overrides.setdefault("mesh_axes", ["data"])
+      overrides.setdefault("ici_data_parallelism", -1)
+
+    if as_argv:
+      return [f"{k}={v}" for k, v in overrides.items()]
+    return overrides
+  except (ImportError, ValueError, TypeError, RuntimeError):  # pragma: no cover - defensive
```
Comment (shralex) [inline tests/utils/test_helpers.py]: Can we narrow down the exception here ? Just in case some legitimate error is thrown that we wouldn't want to hide

File: tests/utils/test_helpers.py (hunk 5 above)
Comment (gulsumgudukbay) [reply tests/utils/test_helpers.py:27]: @shralex  In docs/run_maxtext/decoupled_mode.md it explicitly mentions that tests use get_test_config_path() and that it ensures decoupled_base_test.yml is used in decoupled mode (see around lines 29–32).

File: tests/utils/test_helpers.py (hunk 6 above)
Comment (gulsumgudukbay) [reply tests/utils/test_helpers.py]: narrowed

File: tests/integration/train_tests.py (hunk 4 above)
Comment (gulsumgudukbay) [reply tests/integration/train_tests.py]: there's no API but I added a function to test_helpers for any use in the future.
